components/chassis.py

```diff
@@ -135,33 +140,33 @@ def set(self, desired_state: SwerveModuleState):
         self.state = SwerveModuleState.optimize(self.state, self.get_rotation())
 
         if abs(self.state.speed) < 0.01 and not self.module_locked:
-            self.drive.set(phoenix5.ControlMode.Velocity, 0)
-            self.steer.set(phoenix5.ControlMode.PercentOutput, 0)
+            drive_reqest = phoenix6.controls.VelocityVoltage(0).with_slot(1)
+            self.drive.set_control(drive_reqest)
+
+            steer_request = phoenix6.controls.VoltageOut(0)
+            self.steer.set_control(steer_request)
             return
 
         current_angle = self.get_angle_integrated()
         target_displacement = constrain_angle(
             self.state.angle.radians() - current_angle
         )
         target_angle = target_displacement + current_angle
-        self.steer.set(
-            phoenix5.ControlMode.Position, target_angle * self.STEER_RAD_TO_COUNTS
+        steer_request = (
+            phoenix6.controls.PositionVoltage()
+            .with_velocity(target_angle * self.STEER_RAD_TO_COUNTS)
+            .with_slot(0)
```

Comment:
Isn't slot 0 the default?

If it isn't, shouldn't we be preallocating the control request like CTRE's examples in the docs, and only updating what changes here?

Reply:
It is, thats mb. well this steer_request is redefined by what changes here, the original code changes the control mode and the velocity